### ISSUE TITLE
License issue with c3p0

### ISSUE THREAD

**Issue by danielgratzl** [quartz-scheduler/quartz#19]
Quartz scheduler is released under the Apache 2.0 license but has a dependency on c3p0 (0.9.1.1) which uses LGPL. In my understanding this is not allowed http://www.apache.org/legal/resolved.html#category-x
Newer versions of c3p0 declare a dual license LGPL/EPL which should be be fine to statically link against without violating the Apache license model.

**Comment by jhouserizer** [comment]
We're planning to move to newer c3p0 with the next release.   By the way, that list isn't strictly about license compatibility in general, bur rather what Apache organization advises it's own products to distribute (and Quartz is not an Apache product) - because of differences in terms - not that there's any problem legally with including them together, but that those consuming the product may not like the mixed terms.

**Comment by danielgratzl** [comment]
Alright, thanks for the feedback. Even though i think LGPL would require you to not link the library statically against your code. But I actually do not know enough about this whole license thing, we're just trying to get our dependencies right :)
Thanks for the great work on quartz!

**Comment by jhouserizer** [comment]
See #20

**Comment by msteiger** [comment]
I stumbled over this a few days, because I was surprised to find c3p0 in my classpath despite the fact that I'm using DBCP2 as a connection pool.

I'm surprised to see a JDBC connection pool being explicitly required by a job scheduler. Would it make sense to externalize it into a plugin? Then, users interested in that feature could add `quartz-jdbc-pool` (if that would be its name) instead of `quartz-core` while other might just stick to the plain version and add the pool of their choice and in the version they need.

This is probably linked to #20 - there are always users who want to use a different pool. Having multiple ones floating around isn't ideal imho.

**Comment by zemian** [comment]
@msteiger Our default out of box `org.quartz.utils.PoolingConnectionProvider` uses c3p0 inner class thus the hard dependency. But if you were to provide your own `ConnectionProvider`, then you may exclude the c3p0 from Maven dependency in your project like this:

```
        <dependency>
            <groupId>org.quartz-scheduler</groupId>
            <artifactId>quartz</artifactId>
            <version>${quartz.version}</version>
            <exclusions>
                <exclusion>
                    <groupId>c3p0</groupId>
                    <artifactId>c3p0</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
```

**Comment by jhouserizer** [comment]
won't fix with quartz 2.2.x , c3p0 will be replaced with 2.3.x.
